dotCMS/src/main/java/com/dotcms/security/secret/ServiceIntegrationAPI.java

```diff
@@ -0,0 +1,167 @@
+package com.dotcms.security.secret;
+
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.util.Config;
+import com.liferay.portal.model.User;
+import io.vavr.Tuple2;
+import io.vavr.control.Try;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public interface ServiceIntegrationAPI {
+
+    /**
+     * Given an individual host a list with the associated services is returned.
+     * @param host
+     * @param user
+     * @return
+     */
+    List<String> listServiceKeys(Host host, User user) throws DotDataException, DotSecurityException;
+
+    /**
+     * Conforms a map where the Elements are lists of service unique names, organized by host as key.
+     * @return
+     */
+    Map<String, Set<String>> serviceKeysByHost();
+
+    /**
+     * This returns a json object read from the secret store that contains the service integration configuration and secret.
+     * @param serviceKey the unique service identifier
+     * @param host the host for the respective service key
+     * @param user logged in user
+     * @return
+     */
+     Optional<ServiceSecrets> getSecrets(String serviceKey,
+             Host host, User user) throws DotDataException, DotSecurityException;
+
+    /**
+     * This returns a json object read from the secret store that contains the service integration configuration and secret
+     * This method allows an additional hit against systemHost in case the secret isn't found under the given host
+     * The key is a combination of service + host.
+     * @param serviceKey the unique service identifier
+     * @param fallbackOnSystemHost this param allows  an additional try against system host
+     * @param host the host for the respective service key
+     * @param user logged in user
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    Optional<ServiceSecrets> getSecrets(String serviceKey,
+            boolean fallbackOnSystemHost,
+            Host host, User user) throws DotDataException, DotSecurityException;
+
+    /**
+     * In s similar fashion as `getSecrets` does this method hits the secrets repo but it does not deal or convert the entry into a json object
+     * This only tells you if the service-key exists for a specific host.
+     * @param serviceKey
+     * @param host
+     * @param user
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    boolean hasAnySecrets(final String serviceKey,
```

Comment:
Wondering if this method should throw all these checked exceptions or just wrap into DotRuntime